### PR TITLE
Remove custom block device support.

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Remove page cache file system.
Remove ability to stream files over page cache.
Always use default file system for all copy/transfer operations.
Remove page file read and write channel.

### CODE REVIEW

**Comment by chrisvest** [comment]
`CustomIOConfigValidator` can also be removed.

**Comment by MishaDemianenko** [comment]
@chrisvest oh so you do not want me to keep that custom swapper possibility, because thats why i keep it for now?

**Comment by chrisvest** [comment]
@MishaDemianenko the `CustomIOConfigValidator` is only there for things that don't know how to work with blockdevice and its custom file system. It's a hack and isn't really about the swappers themselves, but about custom file systems that we didn't have a clean way to test for.

**Comment by MishaDemianenko** [comment]
@chrisvest thx, killed!

**Comment by chrisvest** [comment]
I think the failing test can be fixed by deleting it.

**Comment by MishaDemianenko** [comment]
@burqen maybe you also be interested since some work was done here on native indexes as well

**Review by chrisvest — APPROVED** [review]

**Comment by burqen** [comment]
@MishaDemianenko Thanks, it looks good! We could also remove `GBPTreeFileUtil` interface since it only has the trivial file system implementation left.

**Comment by MishaDemianenko** [comment]
@burqen done

File: community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/NativeSchemaIndexAccessor.java [hunk 1]
```diff
@@ -69,7 +69,7 @@
     public void drop() throws IOException
     {
         closeTree();
-        gbpTreeFileUtil.deleteFile( storeFile );
+        fileSystem.deleteFile( storeFile );
```
Comment (burqen) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/NativeSchemaIndexAccessor.java]: Used to be `fileSystem.deleteFileOrThrow` underneath.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/index/labelscan/NativeLabelScanStore.java [hunk 2]
```diff
@@ -412,7 +410,7 @@ private void dropStrict() throws IOException
             index.close();
             index = null;
         }
-        gbpTreeUtil.deleteFile( storeFile );
+        fileSystem.deleteFile( storeFile );
```
Comment (burqen) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/index/labelscan/NativeLabelScanStore.java]: Used to be `fileSystem.deleteFileOrThrow` underneath.

**Review by burqen — CHANGES_REQUESTED** [review]
This _might_ change behaviour. See comments.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/NativeSchemaIndexAccessor.java (hunk 1 above)
Comment (MishaDemianenko) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/NativeSchemaIndexAccessor.java]: oops, that moment when automatic refactoring didn't really work; fixed now

**Review by burqen — APPROVED** [review]

**Comment by burqen** [comment]
@MishaDemianenko Sweet!

**Pull request merged**